### ISSUE TITLE
Look into ways of writing an efficient `divvy`.

### ISSUE THREAD

**Issue by kindaro** [haskell/containers#770]
[There is a function `divvy` in `split`.][1] David @treeowl thinks that we can replicate it in `Data.Sequence` in an efficient way. I am going to look into it.

[1]: https://hackage.haskell.org/package/split-0.2.3.4/docs/Data-List-Split.html#v:divvy

**Comment by treeowl** [comment]
To be very clear, I really only *suspect* we can. The lines between possible and impossible can be subtle. I would personally start by thinking about Louis Wasserman's very clever implementations of `inits` and `tails` (see [my question](https://stackoverflow.com/questions/28906742/how-do-inits-and-tails-work-in-data-sequence) about it and Wasserman's answer). You might or might not also get something from the ideas in `Data.Sequence.chunksOf`.

**Comment by kindaro** [comment]
Yes! A research project, I understand. Good to have some pointers to start from! Thanks!
